GoogleForCreators/web-stories-wp
# Add `<amp-story-auto-analytics>` migration option

## Context



We want to migrate users to the new `<amp-story-auto-analytics>` component, giving them an opt-in if they used the old configuration so far.

## Summary



This PR adds the initial implementation for the `<amp-story-auto-analytics>` migration, behind a feature flag.

See #7739

## Relevant Technical Choices



## To-do



## User-facing changes



**New UI option:**

![Screenshot 2021-09-16 at 22 54 10](https://user-images.githubusercontent.com/841956/133685482-6f25d880-d693-41b6-aa5a-f51ce94c24af.png)


**Demo:**

https://user-images.githubusercontent.com/841956/133685535-e4de6438-c2c8-4dc0-8b70-d50c17afd225.mov


## Testing Instructions






**Note:** before testing, ensure you have a tracking ID set.

**Note** To reset the state between tests, visit `wp-admin/options.php` and set `web_stories_using_legacy_analytics` to `1`. This will make the migration button appear again.


This PR can be tested by following these steps:

1. On a published story, verify that you see `<amp-analytics>` in the source code
1. Enable `Auto Analytics` experiment
2. Go to dashboard settings
3. See "Migrate" button under Analytics section
1. On a published story, verify that you now see `<amp-story-auto-analytics>` in the source code

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #9045
See #7739
